samples/bluetooth/fast_pair/locator_tag/src/main.c

```diff
@@ -353,24 +376,21 @@ static void fmdn_provisioning_state_changed(bool provisioned)
 
 	/* Triggered on the unprovisioning operation. */
 	if (factory_reset_executed) {
+		LOG_INF("The device was unprovisioned and reset to factory settings");
 		LOG_INF("Please press a button to put the device in the Fast Pair discoverable "
 			"advertising mode after a reset to factory settings");
+
 		factory_reset_executed = false;
 
 		return;
 	}
 
-	/* Select the Fast Pair advertising mode according to the FMDN provisioning state. */
-	if (provisioned) {
-		if (is_first_state_changed_cb) {
-			fp_adv_mode = APP_FP_ADV_MODE_NOT_DISCOVERABLE;
-		} else {
-			fp_adv_mode = APP_FP_ADV_MODE_OFF;
-		}
-	} else {
-		fp_adv_mode = APP_FP_ADV_MODE_DISCOVERABLE;
-	}
-	app_fp_adv_mode_set(fp_adv_mode);
+	/* Request the Fast Pair advertising according to the FMDN provisioning state. */
+	app_fp_adv_request(&fp_adv_trigger_clock_sync, provisioned && is_first_state_changed_cb);
+	app_fp_adv_request(&fp_adv_trigger_fmdn_provisioning, false);
+
+	fp_adv_ui_request = !provisioned;
+	app_fp_adv_request(&fp_adv_trigger_ui, fp_adv_ui_request);
 
 	is_first_state_changed_cb = false;
```

Comment:
Should be assigned before early returns

Reply:
This is false as we do not call the `provisioning_state_changed` callback in the `bt_fast_pair_enable` function context. I removed this comment before submitting my review, but somehow it still got submitted.